src/apps/activity_assignments/service.py

```diff
@@ -234,6 +239,54 @@ async def get_all(self, applet_id: uuid.UUID, query_params: QueryParams) -> list
             for schema in schemas
         ]
 
+    async def get_all_by_respondent(
+        self, applet_id: uuid.UUID, respondent_subject_id: uuid.UUID
+    ) -> list[ActivityAssignmentWithSubject]:
+        schemas = await ActivityAssigmentCRUD(self.session).get_by_applet_and_respondent(
+            applet_id, respondent_subject_id
+        )
+
+        respondent_subject = await SubjectsCrud(self.session).get_by_id(respondent_subject_id)
+        if not respondent_subject:
+            raise NotFoundError()
```

Comment:
Should we run this query before querying for the assignments?

Reply:
done